src/Raven.Server/Indexing/TempFileCache.cs

```diff
@@ -214,8 +215,27 @@ public override void SetLength(long value)
         public override void Write(byte[] buffer, int offset, int count)
         {
             var pos = InnerStream.Position;
-            InnerStream.Write(buffer, offset, count);
-            _length = Math.Max(_length, pos + count);
+            try
+            {
+                InnerStream.Write(buffer, offset, count);
+                _length = Math.Max(_length, pos + count);
+            }
+            catch (IOException e)
+            {
+                if (IsDiskFull(e))
+                {
+                    throw new OutOfDiskSpaceException("Out of disk space", e);
```

Comment:
You can use `DiskFullException` instead

Reply:
And we need to provide more info about the out-of-disk space error.
Like in `ThrowDiskFullException` in `VoronIndexOutput`

Reply:
Done